registrations/tests/test_signups_export.py

```diff
@@ -0,0 +1,323 @@
+from typing import Union
+from unittest.mock import patch, PropertyMock
+
+import pytest
+from rest_framework import status
+
+from events.tests.conftest import APIClient
+from events.tests.factories import DataSourceFactory, OrganizationFactory
+from events.tests.utils import versioned_reverse as reverse
+from helevents.tests.factories import UserFactory
+from registrations.tests.factories import (
+    RegistrationUserAccessFactory,
+    SignUpFactory,
+    SignUpGroupFactory,
+)
+
+# === util methods ===
+
+
+def _get_signups_export(
+    api_client: APIClient,
+    registration_id: Union[str, int],
+    file_format: str,
+    query_string: str = None,
+):
+    url = reverse(
+        "registration-signups-export",
+        kwargs={"pk": registration_id, "file_format": file_format},
+    )
+
+    if query_string:
+        url = "%s?%s" % (url, query_string)
+
+    response = api_client.get(url)
+
+    return response
+
+
+def _assert_correct_content(response):
+    assert response.headers["Content-Type"] == (
+        "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
+    )
+    assert response.headers["Content-Disposition"] == (
+        'attachment; filename="registered_persons.xlsx"'
+    )
+    assert len(response.content) > 0
+
+
+def _assert_get_signups_export(
+    api_client: APIClient,
+    registration_id: Union[str, int],
+    file_format: str,
+    query_string: str = None,
+):
+    response = _get_signups_export(
+        api_client, registration_id, file_format, query_string=query_string
+    )
+
+    assert response.status_code == status.HTTP_200_OK
+    _assert_correct_content(response)
+
+    return response
+
+
+def _create_default_signups_data(registration):
+    signup_group = SignUpGroupFactory(registration=registration)
+    SignUpFactory(registration=registration, signup_group=signup_group)
+
+    SignUpFactory(registration=registration)
+    SignUpFactory(registration=registration)
+
+
+# === tests ===
+
+
+@pytest.mark.parametrize(
+    "user_role",
+    [
+        "admin",
+        "regular_user",
+        "regular_user_without_organization",
+        "weakly_identified_registration_user",
+        "admin_of_another_organization",
+        "user_of_another_organization",
+        "anonymous_user",
+    ],
+)
+@pytest.mark.django_db
+def test_signup_export_forbidden_organization_user_roles(
```

Comment:
This test has many conditions. It would be good to separate those conditions into different tests rather than have complex testing logic.

Reply:
I see that there's same like situation in other tests as well. So splitting conditions to own tests or where can parametrize its coolio also 😎

Reply:
True - I'll split these up a bit to simplify the logic